src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -730,6 +732,191 @@ public static nuint RotateRight(nuint value, int offset)
 #endif
         }
 
+        /// <summary>
+        /// Accumulates the CRC (Cyclic redundancy check) checksum.
+        ///
+        /// Uses following Hardware Intrinsics, if supported:
+        /// * uint32_t __crc32b (uint32_t a, uint8_t b)
+        /// * uint32_t __crc32cb (uint32_t a, uint8_t b)
+        /// </summary>
+        /// <param name="crc">The base value to calculate checksum on</param>
+        /// <param name="data">The checksum data</param>
+        /// <returns>The CRC-checksum</returns>
+        [CLSCompliant(false)]
+        public static uint Crc32C(uint crc, byte data)
```

Comment:
Is there anywhere throughout dotnet/runtime these should be used?

Reply:
Unfortunately, I am not sure. I first thought of Zip, but it uses the normal "CRC32" and not the Castagnoli algorithm.

Reply:
There's #44194 putting `CLSCompliant` up for a rethink, but it looks like there hasn't been any hard decisions yet on when it should/shouldn't be used.

Reply:
Thanks for the info, however as long as this is not approved I think we should continue with the current scheme.